src/selectors/selectors.ts

```diff
@@ -135,6 +137,28 @@ export const getPrivateChannelPosts = createSelector([ getSelectedTopicPosts ],
     return posts;
 });
 
-export const getYourPosts = createSelector([ getLocalPosts, getProfile ], (posts, author) => {
-    return posts.filter(post => post.author && post.author.name === author.name);
+export const getYourPosts = createSelector([ getLocalPosts, getPrivatePosts, getProfile ], (posts: Post[], privatePosts: PostListDict, author) => {
+    const yourPubicKey = author.identity!.publicKey;
+    const yourPrivatePosts = Object
+        .values(privatePosts)
+        .reduce((prev, curr) => prev.concat(curr), [])
```

Comment:
It's not clear to me what are you trying to achieve here. It seems like Object.values returns an array, and then in the `reduce` you clone the array. Is this handling some kind of an edge case?

Reply:
`privatePosts` is an Object, the property names are the topics, and their values are list of posts.
Therefore `Object.values` returns a list of lists and the reduce is flatting it out to a single list.

Reply:
I see. I find this alternative easier to read:
```js
const postsByTopics = Object.values(privatePosts);
const yourPrivatePosts = [].concat(...postsByTopics);
```

Reply:
I changed the reduce to concat.

Unfortunately Typescript cannot infer the types correctly so you have to explicitly write out the types.